File: intermediate-server/factory.intermediate.iot/src/main/java/ua/nure/makieiev/factory/intermediate/iot/service/UserService.java
package ua.nure.makieiev.factory.intermediate.iot.service;

import ua.nure.makieiev.factory.intermediate.iot.model.LoginUser;
import ua.nure.makieiev.factory.intermediate.iot.model.UserPlace;
import ua.nure.makieiev.factory.intermediate.iot.model.UserToken;

public interface UserService {

    UserToken signIn(LoginUser loginUser);

    UserPlace findUserPlaceByUserId(long userId, String token);

}

File: mobile/app/src/main/java/ua/nure/makieiev/factory/service/UserService.kt
package ua.nure.makieiev.factory.service

import android.content.Context
import ua.nure.makieiev.factory.model.dto.SignInUserDto
import ua.nure.makieiev.factory.model.dto.SignUpUserDto
import ua.nure.makieiev.factory.model.entity.User

interface UserService {

    fun signUp(signUpUserDto: SignUpUserDto, context: Context): Boolean

    fun signIn(signInUserDto: SignInUserDto, context: Context)

    fun getUserByLogin(login: String, context: Context): User?

}